### ISSUE TITLE
[oracle] 'Found unparsable section' with return %type/%rowtype

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#8071]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting the example below leads to a 'Found unparsable section' error.

### Expected Behaviour

The example should be parsed correctly.
See https://docs.oracle.com/en/database/oracle/oracle-database/26/lnpls/TYPE-attribute.html

### Observed Behaviour

Linting the example below gives the following error:
`Line 3, Position 18: Found unparsable section: '%type;'`
With %rowtype:
`Line 3, Position 18: Found unparsable section: '%rowtype;'`

### How to reproduce

Lint the following example:
```sql
function f_ex(
    p_ex in number
) return ex_tab.a%type;
``` 

```sql
function f_ex(
    p_ex in number
) return ex_tab%rowtype;
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by osarier** [comment]
Just a heads-up: My rowtype example was wrong. I fixed that now.
